### ISSUE TITLE
RuntimeError: ... Default object is null

### ISSUE THREAD

**Issue by hour** [primitiv/primitiv-python#39]
Hello,

I just copied and executed the example code in "primitiv-python/examples/tutorial1_xor.ipynb" and I get error in my machine.
Could you help me to figure out why error?

Code and error message are as following:
```
from graphviz import Source
import numpy as np
from primitiv import *
from primitiv import functions as F

dev = devices.Naive(); Device.set_default(dev)
# if you want to activate CUDA backend, just replacing above to:
#def = devices.CUDA(gpuid); Device.set_default(dev)
g = Graph(); Graph.set_default(g)

w1 = Parameter([8, 2], initializers.XavierUniform())
b1 = Parameter([8], initializers.Constant(0))
w2 = Parameter([1, 8], initializers.XavierUniform())
b2 = Parameter([1], initializers.Constant(0))

---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-24-76722887985e> in <module>()
----> 1 w1 = Parameter([8, 2], initializers.XavierUniform())
      2 b1 = Parameter([8], initializers.Constant(0))
      3 w2 = Parameter([1, 8], initializers.XavierUniform())
      4 b2 = Parameter([1], initializers.Constant(0))

primitiv\_parameter.pyx in primitiv._parameter.Parameter.__init__()

primitiv\_parameter.pyx in primitiv._parameter.Parameter.init()

RuntimeError: C:\Users\...\AppData\Local\Temp\pip-build-7utsq41i\primitiv\primitiv-core\primitiv/mixins.h: 121: Default object is null.
```
Thanks in advance,
Hour

**Comment by odashi** [comment]
This problem looks very like primitiv/primitiv#47.
Are you using primitiv on Windows? We have investigated this issue on Linux/Mac/FreeBSD, but no observation about Windows so far...
If you can provide us your time, could you take a look at primitiv/primitiv#47 and provide some information on your environment related to this issue?

@vbkaisetsu, could you work to resolve this issue?

**Comment by odashi** [comment]
Option: you may be able to avoid this issue by directly specifying the device and graph to functions that needs them:
```python
w1 = Parameter(shape, initializer, device=dev)
```

**Comment by vbkaisetsu** [comment]
I will try to resolve this issue on the weekend.
@hour Could you give some your environment information? VC++, MinGW, ... and a version number.

**Comment by hour** [comment]
@vbkaisetsu
I used
- Window 10 - 64 bits
- VC++ 2017 version 15.7 v14.14 latest v141 tools
- Microsoft Visual Studio 2017
- Python 3.6 - 64 bits (installed for its official website, not from Microsoft Visual Studio installer)
- MinGW (I am not sure I have it installed), How can I check if it is installed (or its version) in my machine?

Please let me know if you need more information.

Hour

**Comment by hour** [comment]
@odashi with "w1 = Parameter(shape, initializer, device=dev)", I can skip the error. But I get the same error on other code as following.

```
def build_graph():
    # Input
    # F.input() takes an ndarray, or a list of ndarrays.
    x = F.input(inputs)
    
    # Calculating outputs by 2-layer perceptron.
    h = F.tanh(F.parameter(w1) @ x + F.parameter(b1))
    y = F.tanh(F.parameter(w2) @ h + F.parameter(b2))
    
    # Ground truth
    # F.raw_input() takes a shape and a corresponding list of
    # raw float values with column-major order.
    t = F.raw_input(Shape([1], 4), labels)
    
    # Calculating loss by squared error.
    diff = y - t
    loss = F.batch.sum(diff * diff)
    
    return y, loss

g.clear()
build_graph()
Source(g.dump("dot"))


---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-48-35a2d8138021> in <module>()
      1 g.clear()
----> 2 build_graph()
      3 Source(g.dump("dot"))

<ipython-input-47-d996ae6c88ee> in build_graph()
      2     # Input
      3     # F.input() takes an ndarray, or a list of ndarrays.
----> 4     x = F.input(inputs)
      5 
      6     # Calculating outputs by 2-layer perceptron.

primitiv\_function.pyx in primitiv._function.functions.input()

primitiv\_function.pyx in primitiv._function.functions.raw_input()

RuntimeError: C:\Users\...\AppData\Local\Temp\pip-build-7utsq41i\primitiv\primitiv-core\primitiv/mixins.h: 121: Default object is null.

```

**Comment by hour** [comment]
I checked in the directory C:\Users\...\AppData\Local\Temp\*.
There is no "pip-build-7utsq41i".

I guess maybe it's not generated during primitive installation.

**Comment by odashi** [comment]
Oh sorry, you also need to specify device and graph for some cases.
If the new node returned by `F` functions has no parent nodes (such as `F.input`), you have to specify device and graph to inform primitiv where you put information corresponding to the nodes. If it has parents (such as `+`, `F.matmul`), the device and graph are inherited from parents automatically.
Please grab [_function.pyx](https://github.com/primitiv/primitiv-python/blob/develop/primitiv/_function.pyx) for details of each function.
The default device/graph functionality is originally introduced to avoid passing the same objects to primitiv repeatedly.

**Comment by vbkaisetsu** [comment]
I tried primitiv/primitiv#47 but it does not fix this problem.

**Comment by vbkaisetsu** [comment]
C++'s set_default() works correctly on Windows.

**Comment by odashi** [comment]
Yes, this is obviously an issue related to the Python APIs beyond DLLs. C++'s APIs are self-contained so it should work.

**Comment by odashi** [comment]
I guessed that it is better to separate code of the Python's default objects functionality from the C++ API. Current implementation requires the linker to merge the same static member definitions in multiple DLLs that basically looks extraordinary although the C++ standard allows it.

**Comment by odashi** [comment]
This issue may be fixed by #41. Please try it again.
